lib/services/presets.go

```diff
@@ -29,6 +29,60 @@ import (
 	"github.com/gravitational/teleport/lib/modules"
 )
 
+func NewPresetAutomaticAccessApproverRole() types.Role {
+	enterprise := modules.GetModules().BuildType() == modules.BuildEnterprise
+	if !enterprise {
+		return nil
+	}
+	role := &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V7,
+		Metadata: types.Metadata{
+			Name:        teleport.PresetAutomaticAccessApprovalRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Approves any access request",
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				ReviewRequests: &types.AccessReviewConditions{
+					Roles: []string{"*"},
+				},
+			},
+		},
+	}
+	return role
+}
+
+// NewPresetAutomaticAccessBotUser returns a new User that has (via the
+// the `PresetAutomaticAccessApprovalRoleName` role) the right to automatically
+// approve any access requests.
+//
+// This user must not:
+//   - Be allowed to log into the cluster
+//   - Show up in user lists in any UI (including Web, `tsh`, `tctl`, etc)
+//
+// TODO(tcsc): Implement/enforce above restrictions on this user
+func NewPresetAutomaticAccessBotUser() types.User {
+	enterprise := modules.GetModules().BuildType() == modules.BuildEnterprise
+	if !enterprise {
+		return nil
+	}
+
+	user := &types.UserV2{
+		Kind:    types.KindUser,
+		Version: types.V2,
+		Metadata: types.Metadata{
```

Comment:
Can we set some labels on the user indicating this is a builtin "system" user. @mdwn is adding similar label on reviewer/requester role in https://github.com/gravitational/teleport/pull/27673, let's settle on something e.g. `teleport.internal/system: true` and use it everywhere.

Reply:
That was my plan for filtering, etc. Happy to add it to this PR instead the filtering one.